lib/primo/search.rb

```
@@ -51,7 +51,18 @@ def self.get(params = {})
       params = params.to_h.transform_keys(&:to_sym)
       method = get_method params
       (url, query) = url(params)
-      new super(url, query:, timeout: Primo.configuration.timeout(params)), method
+
+      retry_count = Primo.configuration.enable_retries ? Primo.configuration.retries : 0
+      begin
+        new super(url, query:, timeout: Primo.configuration.timeout(params)), method
+      rescue Net::ReadTimeout
+        if (Primo.configuration.retries && (retry_count -= 1) > 0)
+          Primo.configuration.logger.warn("Primo request timed out. Retrying. [#{retry_count}]")
+          retry
+        else
+          raise "Primo request timed out"
```

You don't need an else statement to raise an error just let the it fail again.